### PR TITLE
Separate log debug into a specific file, not mandatory

### PR DESCRIPTION
**Pull request by renaldihusintkpd**
I want to make this changes for easing monitoring process which is hard to define the log and if we merge access log and debug log the size of log is too big.

### CODE REVIEW

File: init.go [hunk 1]
```diff
@@ -28,6 +32,7 @@ var Debug *log.Logger
 func init() {
 	flag.StringVar(&stdoutLog, "l", "", "log file for stdout")
 	flag.StringVar(&stderrLog, "e", "", "log file for stderr")
+	flag.StringVar(&debugLog, "d", "", "log file for debug") // only if want to use debugLog
```
Comment (bmishra) [inline init.go]: Use value stored here for debug file name instead of `os.Getenv("DEBUG_LOG")` below.

File: init.go [hunk 2]
```diff
@@ -70,7 +75,15 @@ func LogInit() {
 func SetDebug(enabled bool) {
 	if enabled {
 		debugFlag = true
-		Debug = log.New(os.Stdout, "debug:", log.Ldate|log.Ltime|log.Lshortfile)
+		if filename := os.Getenv("DEBUG_LOG"); filename != "" {
+			file, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
```
Comment (bmishra) [inline init.go]: can we use `reopen()`, update it to not call `dup2()` if fd is not passed.

**Review by bmishra — CHANGES_REQUESTED** [review]

**Comment by bmishra** [comment]
Add/Update PR description to explain what/why you are making the changes.

File: init.go (hunk 2 above)
Comment (renaldihusintkpd) [reply init.go]: yes, i had make the changes as requested.

File: init.go (hunk 1 above)
Comment (renaldihusintkpd) [reply init.go]: Okay, i had change the code using value stored. Please let me know if there is something to be fix again.

**Review by bmishra — APPROVED** [review]

**Pull request merged**